File: implement-strstr/sol.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle:
            return 0
        if not haystack:
            return -1

        s1 = len(haystack)
        s2 = len(needle)
        if s1 < s2:
            return -1

        return haystack.find(needle)

File: implement-strstr/sol.go
package main

import "strings"

func strStr(haystack string, needle string) int {
	if needle == "" {
		return 0
	}
	if haystack == "" {
		return -1
	}
	if len(haystack) < len(needle) {
		return -1
	}
	return strings.Index(haystack, needle)
}
